src/tools/launcher/launcher.cc

```diff
@@ -100,79 +99,31 @@ static bool FindManifestFileImpl(const wchar_t* argv0, wstring* result) {
   return false;
 }
 
-wstring BinaryLauncherBase::FindManifestFile(const wchar_t* argv0) {
-  wstring manifest_file;
-  if (!FindManifestFileImpl(argv0, &manifest_file)) {
-    return L"";
-  }
-  // The path will be set as the RUNFILES_MANIFEST_FILE envvar and used by the
-  // shell script, so let's convert backslashes to forward slashes.
-  std::replace(manifest_file.begin(), manifest_file.end(), '\\', '/');
-  return manifest_file;
-}
-
 wstring BinaryLauncherBase::GetRunfilesPath() const {
   wstring runfiles_path =
       GetBinaryPathWithExtension(this->commandline_arguments[0]) + L".runfiles";
   std::replace(runfiles_path.begin(), runfiles_path.end(), L'/', L'\\');
   return runfiles_path;
 }
 
-void BinaryLauncherBase::ParseManifestFile(ManifestFileMap* manifest_file_map,
-                                           const wstring& manifest_path) {
-  ifstream manifest_file(AsAbsoluteWindowsPath(manifest_path.c_str()).c_str());
-
-  if (!manifest_file) {
-    die(L"Couldn't open MANIFEST file: %s", manifest_path.c_str());
-  }
-
-  string line;
-  while (getline(manifest_file, line)) {
-    size_t space_pos = line.find_first_of(' ');
-    if (space_pos == string::npos) {
-      die(L"Wrong MANIFEST format at line: %s", line.c_str());
-    }
-    wstring wline = blaze_util::CstringToWstring(line);
-    wstring key = wline.substr(0, space_pos);
-    wstring value = wline.substr(space_pos + 1);
-    manifest_file_map->insert(make_pair(key, value));
-  }
-}
-
-wstring BinaryLauncherBase::Rlocation(const wstring& path,
-                                      bool need_workspace_name) const {
+wstring BinaryLauncherBase::Rlocation(wstring path,
+                                      bool has_workspace_name) const {
   // No need to do rlocation if the path is absolute.
   if (blaze_util::IsAbsolute(path)) {
     return path;
   }
 
-  // If the manifest file map is empty, then we're using the runfiles directory
-  // instead.
-  if (manifest_file_map.empty()) {
-    wstring query_path = runfiles_dir;
-    if (need_workspace_name) {
-      query_path += L"/" + this->workspace_name;
-    }
-    query_path += L"/" + path;
-    return query_path;
+  if (path.find(L"external/") == 0) {
```

Comment:
What if users has an `external` directory in their source tree? This could be wrong. Maybe `path.startwith(L"external/")` is better? Or just don't check it at all, because `<workspace_name>/external/<external_repo_name><file_path_in_external_repo>` also exits in manifest file.